### ISSUE TITLE
Errors on both Gulp cmd, and on generator

### ISSUE THREAD

**Issue by bradsturtevant** [dustinspecker/generator-ng-poly#109]
I generally know what I doing with Node/Grunt Node/Gulp but am getting errors on all commands in ng-poly. Have it installed/working on other computers. Is NodeJS 12 a problem?

Start of Error Message is: 
C:\Users\Brad\devel\src\uvo\node_modules\gulp-less\index.js:68
    }).done(undefined, cb);
       ^
TypeError: undefined is not a function
  at DestroyableTransform._transform (C:\Users\HIS93301\devel\src\uvo\node_modules\gulp-less\index.js:68:8)

**Comment by bradsturtevant** [comment]
On NodeJS 10.35 I can generate a route (from default gen code) from > app/home but not > app module. Is the a real module, or should I generate a module first?

$ yo ng-poly:route myVehicles
? Which module is this for? app
? What's the URL for this route? /my-vehicles
? What's the templateURL for this route? /my-vehicles.tpl.html

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: ENOENT, no such file or directory 'c:\Users\Brad\devel\src\swks-genpoly\app\swks-genpoly.js'
    at Object.fs.openSync (fs.js:439:18)
    at Object.fs.readFileSync (fs.js:290:15)
    at writing (c:\Users\Brad\Software\nodejs\node_modules\generator-ng-poly\route\index.js:66:17)
    at c:\Users\Brad\Software\nodejs\node_modules\generator-ng-poly\node_modules\yeoman-generator\lib\base.js:408:16
    at processImmediate [as _immediateCallback] (timers.js:354:15)

$ yo ng-poly:route myVehicles
? Which module is this for? app\home
? What's the URL for this route? /my-vehicles
? What's the templateURL for this route? home/my-vehicles.tpl.html
   create e2e\my-vehicles\my-vehicles.po.js
   create e2e\my-vehicles\my-vehicles_test.js
   create app\home\my-vehicles-controller.js
   create app\home\my-vehicles-controller_test.js
   create app\home\my-vehicles.tpl.html
   create app\home\my-vehicles.less

**Comment by dustinspecker** [comment]
Hi, some of the Gulp plugins aren't working with Node 0.12. I know gulp-sass for sure isn't working, not sure on the others. We should add a note to the README specifying 0.10 for now.

That route thing is a bug - sort of. We definitely strongly encourage creating a new module and adding a route to that. We really shouldn't be adding components to the root module.

The reason this bug is happening is because if someone specifies to add to the app.js module, then we get the application name from package.json because that's the actual name of the Angular module within app.js. This will work for all components added to app.js module, since they need to just know the name of the module. The route subgenerator attempts to open the module file (which we assume is the name of the module, since it is in every other case), but since app isn't the name of the actual module ng-poly fails on this. I'll look into this later, but we don't encourage adding routes to the root module.

Thank you for catching these issues.

**Comment by bradsturtevant** [comment]
Ok! Thanks for your quick response (me just back from long bike ride). The NodeJS 12 behavior for SASS is a bit disconcerting--ask for SASS but get LESS in bower :-)  I have always asked for LESS in the past and thought, hmmm. And as I tried to mention the Gulp commands in NodeJS 12 such as build, dev give fatal errors.

Thanks again for your support. I am contractor, and just sent out lengthy email priasing generator-ng-poly and Gulp over Grunt. Then testing, and yikes! But everything makes sence from your reply. I do read README.md and notes are always an important way to not get emails like this.

I found a like to highest version is 10.x series (with some difficulty). Might be nice to add a link in README.md to this latest 10.x

http://blog.nodejs.org/2014/06/16/node-v0-10-29-stable/

Thanks,
Brad
